pkg/utils/utils_test.go

```diff
@@ -233,6 +234,77 @@ func TestConvertKeys(t *testing.T) {
 	}
 }
 
+func TestRevertKeys(t *testing.T) {
+	type args struct {
+		encodingStrategy esv1beta1.ExternalSecretConversionStrategy
+		decodingStrategy esv1alpha1.PushSecretConversionStrategy
+		in               map[string][]byte
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    map[string][]byte
+		wantErr bool
+	}{
+		{
+			name: "revert unicode extended",
+			args: args{
+				encodingStrategy: esv1beta1.ExternalSecretConversionUnicode,
+				decodingStrategy: esv1alpha1.PushSecretConversionReverseUnicode,
+				in: map[string][]byte{
+					"😀foo😁bar😂baz😈bing": []byte(`noop`),
+				},
+			},
+			want: map[string][]byte{
+				"😀foo😁bar😂baz😈bing": []byte(`noop`),
```

Comment:
Not going to lie, this was quite confusing as to what's happening. :D As somehow who doesn't really mess around with unicode it wasn't apparent that it translates unicode to parsed unicode.

A small description of what this test does would go a long way, or at least in the description of the function that you added into util should explain what it does and why.

Reply:
I know you added
>Replaces encoded characters back to the original value depending on convert strategy.

I would expand this with saying something like, the unicode value is converted into the actual unitcode character based on the encoding strategy.

Or something like that. :D Thanks. :)

Reply:
yes, this is confusing indeed, I tried to apply your suggestions and rephrased the function and the tests description